### ISSUE TITLE
Should this fork become its own Project?

### ISSUE THREAD

**Issue by allusive-dev** [allusive-dev/compfy#34]
Hey Everyone,
Firstly I want to say thank you for all the support recently. 
As of posting this we have nearly reached 100 stars on GitHub and many more downloads on Arch and other systems!

I wanted to bring up this discussion since recently there has been pressure to merge my changes with upstream, though that was never my plan it seems like it could be considered inevitable since this project is still `picom`.

That is why I have brought this up, I am considering turning this into its own project no longer directly related to picom. Though I would still plan to bring in stable updates from `yshui/picom`  and of course still provide credit to contributors to that package.

This would include a rename of the project (new name is still not decided, (will probably make it a community post)), and some other large refactors.

This is a big step for this project of mine so I want to hear everyone's opinions on whether this is the way forward.

**Comment by Mauladen** [comment]
At this stage you could make changes to picom so that as many people as possible can use your work, but if you are sure that you can continue to support the project and develop it, you can safely do it independently of picom

**Comment by allusive-dev** [comment]
> If you are sure that you can continue to support the project and develop it, you can safely do it independently of picom

I believe i can yes. Soon I will be getting alot more free time aswell so the transition should be very smooth should I go ahead with it.

**Comment by allusive-dev** [comment]
> But as you realize the main risk is that separate forks of the main project die sooner or later and it's very sad.

Yeah. My hope is that this project is not only an alternative to Picom, but also a successor!

> I just wish you good luck and want to say thank you for the work you have done

Thank you, it genuinely means a lot. <3

**Comment by allusive-dev** [comment]
Quick update. New name will be `Compfy` if everything goes ahead as suggested!

**Comment by naan187** [comment]
> I wanted to bring up this discussion since recently there has been pressure to merge my changes with upstream, though that was never my plan it seems like it could be considered inevitable since this project is still picom.

I myself would really like if all of the amazing work you are doing here is made available to the most people possible by submitting the changes upstream. Although since (I assume) this fork (for now) makes some big breaking changes, I'm not sure how well that would play out. You'd probably have to discuss and work some stuff out with yshui to get your changes implemented.

> Yeah. My hope is that this project will be not only an alternative to Picom, but also a successor!

Well, good luck with that!

**Comment by allusive-dev** [comment]
> I myself would really like if all of the amazing work you are doing here is made available to the most people possible by submitting the changes upstream.

I have already had a discussion with some who have write access to upstream and at the moment it would not be possible to merge with upstream at any point because the git history is not properly setup on this repository and another few issues.

**Comment by allusive-dev** [comment]
Update!
This thread will remain open but for those with Matrix we now have a chat room for discussion and support.
https://matrix.to/#/#compfy:matrix.org

**Comment by SolninjaA** [comment]
Just adding my opinion here. Personally, I think if these changes are added to upstream many more people will be able to enjoy the features of this fork. Right now, a lot of people use the upstream either because they do not know forks exist or they assume the forks are unstable etc.

I think that merging with the upstream will make this project continue in a more "open source" and "free software" way.
However, that being said, I think merging should only happen if yshui decides that they like your features and features to come. I would hate to see features stopped from being implemented because yshui decides against it.

From what I can tell, merging with the upstream will bring this project to more people in the world and as I said I feel this would coincide most with the "open source" and "free software" movements. But, making this project separate would provide more freedom to add desired features.

Personally, I think both of these options aren't desirable and that this project should remain as it always has been, at least for a bit longer. Hopefully this would shed more light on the opinions of yshui regarding the merge and other important factors.

**Comment by allusive-dev** [comment]
@SolninjaA 
Thank you for your input.
I guess I need to clarify this now but there is no chance this repository will be merged with upstream as I have discussed with ones with write access. This is because I had not setup correct git history when starting this repository and because of a incredibly large amount of styling changes I have made.
Because of these I will not ever be able to merge with upstream basically.

Compfy will not be fully released until a few more bugs and issues are fixed aswell as some ease-of-use changes will be released soon. These involve fixing and overhauling the manpages, improving console outputs, improving configuration syntax and more code cleanup.

Once I am sure Compfy is ready for a full release and it's finally split away from Picom I will make an announcement.
